Sprint 7 - final task 1. added file manager 2. added save and load from file functions 3. added test coverage for file manager

imposmas opened the pull request.

imposmas committed:
Sprint 7 - final task
1. added file manager
2. added save and load from file functions
3. added test coverage for file manager

imposmas committed:
checkstyle correction

imposmas committed:
checkstyle correction

imposmas committed:
checkstyle correction

imposmas committed:
deleted afterEach from tests - not needed

VadimZharkov commented on src/ru/yandex/practicum/kanban/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,219 @@
+            System.out.println("Файл не найден. Будет создан новый");
+            return new FileBackedTaskManager(file);
+        } catch (IOException e) {
+            System.out.println("Произошла ошибка во время чтения файла.");
```
Здесь лучше выбросить FileSaveException т.к. продолжать дальше не имеет смысла

VadimZharkov commented on src/ru/yandex/practicum/kanban/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,219 @@
+            } else if (taskType.equals("SUBTASK")) {
+                SubTask subTask = new SubTask();
+                subTask = subTask.fromCSVStringFormat(lines.get(i));
+                if (!subTasks.keySet().contains(subTask.getId())) {
```
Можно не проверять, т.к. если такой ключ уже каким-то образом есть, то subTasks.put просто перезапишет значение в мапе.

VadimZharkov commented on src/ru/yandex/practicum/kanban/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,219 @@
+            } else if (taskType.equals("EPIC")) {
+                Epic epic = new Epic();
+                epic = epic.fromCSVStringFormat(lines.get(i));
+                if (!epics.keySet().contains(epic.getId())) {
```
См коммент выше

VadimZharkov commented on src/ru/yandex/practicum/kanban/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,219 @@
+                if (!epics.keySet().contains(epic.getId())) {
+                    epics.put(epic.getId(), epic);
+                }
+            }
```
Нужно еще идентификаторы подзадач добавить в эпики

VadimZharkov commented on test/ru/yandex/practicum/kanban/managers/FileBackedTaskManagerTest.java:
```diff
@@ -0,0 +1,102 @@
+        int taskId3 = fileTaskManager.addNewTask(task3);
+
+        List<String> lines = readLinesFromFile(FILE_PATH.toFile());
+        assertTrue(lines.get(0).equals(FileConstants.CSV_HEADER));
```
еще бы добавить assert с проверкой lines на null

VadimZharkov reviewed (changes requested):
В целом хорошая работа, посмотри, что нужно исправить и можно улучшить.

imposmas committed:
1. added writing/reading epic's subtasks to/from file
2. corrected loading tasks from file to avoid additional check
3. added test to check if we can load file that returns empty lines (null)
